src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -362,9 +369,13 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         // remove the last segment as we only need to place separators in between the parts of the words
         segs.pop();
         fluid.each(segs, function (seg) {
-            var separator = $(separatorMarkup)[0];
-            node = node.splitText(seg.length);
-            node.parentNode.insertBefore(separator, node);
+            // Do not attempt to process empty segs, which could lead to an infinite recursion.
+            // See: https://issues.fluidproject.org/browse/FLUID-6554
+            if (seg.length) {
```

Comment:
Might an even safer test do a no-op unless there are 2 or more segments? 
Also I think some of the other logic in this function could be better explained - 
````
        // remove \u200B characters added hyphenateText
        var hyphenated = hyphenator.hyphenateText(node.textContent).replace(/\u200B/gi, "");
````
How do these characters get there, and what are the consequences if they are not removed?